### DISCUSSION TITLE
Preserving and Restoring Column Visibility Changes in a Table Column Using Local Storage

### DISCUSSION THREAD

**Discussion by SinaMAlizadeh** [KevinVandy/mantine-react-table#179] [General]
How can I save changes to column visibility in local storage and restore them when returning to the page and table? I would like to preserve the user's actions regarding column visibility changes and display a history of their selections, so that they can see the columns they had chosen previously. Is there a way to track and retrieve table visibility changes each time they occur?

**Comment by agex-n7** [comment]
You can manage the visibility state yourself with the help of `onColumnVisibilityChange` and `state.columnVisibility`. You can wire that up to any state management solution you fancy.

About column visibility:
https://www.mantine-react-table.com/docs/api/state-options#columnVisibility-state-option
https://tanstack.com/table/v8/docs/api/features/column-visibility
